### ISSUE TITLE
Interacting with AlertView has an issue on iOS 15.2 Simulators and Xcode-13.2.beta Failed With AppiumServiceBuilder

### ISSUE THREAD

**Issue by mochitest** [appium/appium#16292]
## The problem
auto accept alert on iOS doesn't work using AppiumServiceBuilder to start the appium.

It works when using node . to start appium
I have built the latest appiumv1.23.0-beta.0 version from git.

## Environment

Appium version (or git revision) that exhibits the issue: Appium v1.23.0-beta.0
Last Appium version that did not exhibit the issue (if applicable):
Desktop OS/version used to run Appium: macOS Big Sur 11.6
Node.js version (unless using Appium.app|exe): v16.13.0
Npm or Yarn package manager: 0.37.2
Mobile platform/version under test: iOS 15.2
Real device or emulator/simulator: iPhone 8 simulator

## Details

There is a modal dialog shown on a fresh app installation which is prompting to Allow Sending Notifications...
I've to accept popup in order to unblock test flow. It is working when I start the appium with node command
But does not work if I am using appiumServiceBuilder
AppiumDriverLocalService server = AppiumDriverLocalService.buildService(serviceBuilder);

## Link to Appium logs

org.openqa.selenium.WebDriverException: An unknown server-side error occurred while processing the command. Original error: *** -[__NSArrayM insertObject:atIndex:]: object cannot be nil

Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:17:03'
testImplementation group: 'io.appium', name: 'java-client', version: '7.6.0'

## Code To Reproduce Issue [ Good To Have ]

Please remember that with sample code it's easier to reproduce the bug and it's much faster to fix it.

Please git clone https://github.com/appium/appium and from the `sample-code` directory, use one of your favourite languages and sample apps to reproduce the issue.

In case a similar scenario is missing in sample-code, please submit a PR with one of the sample apps provided.

**Comment by mykola-mokhnach** [comment]
Duplicate of https://github.com/appium/appium/issues/16025

Please make sure the correct/latest version of WebDriverAgent is running on the device under test.
Please provide the server log as well.

**Comment by mochitest** [comment]
021-12-31 16:45:25:325 [W3C (7599caac)] Driver proxy active, passing request on via HTTP proxy
2021-12-31 16:45:25:325 [XCUITest] Executing command 'proxyReqRes'
2021-12-31 16:45:25:326 [WD Proxy] Matched '/wd/hub/session/7599caac-ab2a-4c97-804f-5e6786718415/element/68000000-0000-0000-3E25-000000000000/click' to command name 'click'
2021-12-31 16:45:25:327 [WD Proxy] Proxying [POST /wd/hub/session/7599caac-ab2a-4c97-804f-5e6786718415/element/68000000-0000-0000-3E25-000000000000/click] to [POST http://127.0.0.1:6900/session/D7C79A72-91F6-4599-915D-28C0615C5415/element/68000000-0000-0000-3E25-000000000000/click] with body: {}
2021-12-31 16:45:25:545 [WD Proxy] Got response with status 500: {"value":{"error":"unknown error","message":"*** -[__NSArrayM insertObject:atIndex:]: object cannot be nil","traceback":"(\n\t0   CoreFoundation                      0x000000010cf4dba4 __exceptionPreprocess + 242\n\t1   libobjc.A.dylib                     0x000000010cc68be7 objc_exception_throw + 48\n\t2   CoreFoundation                      0x000000010cfcdf64 _CFThrowFormattedException + 194\n\t3   CoreFoundation                      0x000000010cfcbbf9 -[__NSArrayM insertObject:atIndex:].cold.2 + 0\n\t4   CoreFoundation                      0x000000010ce49aee -[__NSArrayM insertObject:atIndex:] + 1095\n\t5   XCUIAutomation                      0x000000012020d59f __104-[XCAXClient_iOS interruptingUIElementsAffectingSnapshot:checkForHandledElement:containsHandledElement:]_block_invoke + 1362\n\t6   XCTestCore                          0x000000011fd8e9ad -[XCTContext _runActivityNamed:type:block:] + 245\n\t7   XCTestCore                          0x000000011fd8db70 +[XCTContext runInternalActivityNamed:block:]...
2021-12-31 16:45:25:560 [W3C] Matched W3C error code 'unknown error' to UnknownError
2021-12-31 16:45:25:575 [HTTP] <-- POST /wd/hub/session/7599caac-ab2a-4c97-804f-5e6786718415/element/68000000-0000-0000-3E25-000000000000/click 500 251 ms - 764
@mykola-mokhnach

**Comment by mochitest** [comment]
{"testmanagerdVersion":28,"name":"iOS","sdkVersion":"15.2","version":"15.2"},"ios":{"simulatorVersion":"15.2"},"ready":true,"build":{"upgradedAt":"499162500000","time":"Dec 29 2021 10:59:17","productBundleIdentifier":"com.facebook.WebDriverAgentRunner"}},"sessionId":null}

**Comment by mochitest** [comment]
close this issue due to old WebDriverAgent
